### DISCUSSION TITLE
How to get extended records in hooks?

### DISCUSSION THREAD

**Discussion by SunPodder** [pocketbase/pocketbase#3030] [Q&A]
```js
// frontend
let res = await pb.collection("conversations").getFullList({
    expand: 'users'
});

// backend
onRecordsListRequest((e) => {
    if (e.httpContext.get("admin")) return;

    console.log(JSON.stringify(e.records, null, 4));
}, "conversations");
```
I'm calling this on the frontend, but on the backend I don't get any expanded data though the expanded data is returned to the client. Also as we don't have access to the request, I cannot know whether any expand was requested, so I could expand it manually.

**Comment by ganigeorgiev** [comment]
The records enriching is done as final step, right before returning the response to the client.

> Also as we don't have access to the request...

The request context is available under `e.httpContext`.

Could you elaborate what are you trying to do? Maybe a custom route or [view collection](https://pocketbase.io/docs/collections/#view-collection) could be better for your use case?

**Comment by SunPodder** [comment]
> The request context is available under e.httpContext.

it shows empty here :\
![image](https://github.com/pocketbase/pocketbase/assets/73415680/7950605c-d7b0-49c9-9ca7-6bf0818eaff1)

I'm trying to assign some default values to the fields as described here #3027. Anyway, I think doing this on the frontend will be better for now, but it would save a lot code repeating for me if can be done here.

Also maybe calling these hooks at the last step would be better.

**Reply by ganigeorgiev** [reply]
> it shows empty here...

Because the `httpContext` object doesn't have public fields that can be serialized.
For more details about the available fields and methods you can find in https://pocketbase.io/jsvm/interfaces/echo.Context.html.
